From 3469430b8025d7b76e8db9271838c020bd32c783 Mon Sep 17 00:00:00 2001
From: Julien Gilli <julien.gilli@joyent.com>
Date: Thu, 8 Feb 2018 15:03:24 -0800
Subject: [PATCH] TRITON-118 vmapi should only look for "ready" volumes when
 checking volumes reachability Reviewed by: Nick Zivkovic
 <zivkovic.nick@gmail.com> Approved by: Nick Zivkovic
 <zivkovic.nick@gmail.com>

---
 lib/endpoints/vms.js | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/endpoints/vms.js b/lib/endpoints/vms.js
index 1c25c7e..33c09ec 100644
--- a/lib/endpoints/vms.js
+++ b/lib/endpoints/vms.js
@@ -1332,7 +1332,8 @@ function createVm(req, res, next) {
 
                 req.app.volapi.listVolumes({
                     name: volumeName,
-                    owner_uuid: ownerUuid
+                    owner_uuid: ownerUuid,
+                    state: 'ready'
                 }, function onListVols(listVolsErr, vols) {
                     var checkReachableErr;
                     var foundVolNetwork = false;
-- 
2.21.0

